crates/accelerate/src/target_transpiler/mod.rs

```diff
@@ -0,0 +1,1285 @@
+                                            x.index()
+                                        }
+                                    }) + 1,
+                                ));
+                            qarg_obj = Some(qarg.clone())
+                        }
+                        qargs_val.insert(qarg_obj.to_owned(), properties[qarg].clone());
+                        self.qarg_gate_map
+                            .entry(qarg_obj)
+                            .and_modify(|e| {
+                                if let Some(e) = e {
+                                    e.insert(name.clone());
+                                }
+                            })
+                            .or_insert(Some(HashSet::from([name.clone()])));
+                    }
+                }
+            }
+        }
+        // TODO: Modify logic once gates are in rust.
+        self._gate_name_map.insert(name.clone(), instruction);
+        self.gate_map.insert(name, qargs_val);
+        self.non_global_basis = None;
+        self.non_global_strict_basis = None;
+        Ok(())
+    }
+
+    /// Update the property object for an instruction qarg pair already in the Target
+    ///
+    /// Args:
+    ///     instruction (str): The instruction name to update
+    ///     qargs (tuple): The qargs to update the properties of
+    ///     properties (InstructionProperties): The properties to set for this instruction
+    /// Raises:
+    ///     KeyError: If ``instruction`` or ``qarg`` are not in the target
+    #[pyo3(text_signature = "(instruction, qargs, properties, /,)")]
+    fn update_instruction_properties(
+        &mut self,
+        instruction: String,
+        qargs: Option<Qargs>,
+        properties: Option<InstructionProperties>,
+    ) -> PyResult<()> {
+        if !self.contains_key(&instruction) {
+            return Err(PyKeyError::new_err(format!(
+                "Provided instruction: '{:?}' not in this Target.",
+                &instruction
+            )));
+        };
+        let mut prop_map = self[&instruction].clone();
+        if !(prop_map.contains_key(&qargs)) {
+            return Err(PyKeyError::new_err(format!(
+                "Provided qarg {:?} not in this Target for {:?}.",
+                &qargs.unwrap_or_default(),
+                &instruction
+            )));
+        }
+        prop_map.entry(qargs).and_modify(|e| *e = properties);
+        self.gate_map
+            .entry(instruction)
+            .and_modify(|e| *e = prop_map);
+        Ok(())
+    }
+
+    /// Get the qargs for a given operation name
+    ///
+    /// Args:
+    ///     operation (str): The operation name to get qargs for
+    /// Returns:
+    ///     set: The set of qargs the gate instance applies to.
+    #[pyo3(name = "qargs_for_operation_name")]
+    fn py_qargs_for_operation_name(&self, operation: String) -> PyResult<Option<Vec<Qargs>>> {
+        match self.qargs_for_operation_name(&operation) {
```

Comment:
I think this should work: 
```suggestion
    fn py_qargs_for_operation_name(&self, operation: &str) -> PyResult<Option<Vec<Qargs>>> {
        match self.qargs_for_operation_name(operation) {
```